### PR TITLE
Added cookie validation whitelist

### LINKED ISSUE #9158: Cookie could not be read
I'm setting-up a cookie before facebook login, then in the callback function I'm trying to retrieve it.

It is accessible through $_COOKIE, but not with Yii::$app->getRequest()->getCookies()->getValue().

Example:
```
$cookie1 = Yii::$app->getRequest()->getCookies()->getValue('actionAfterLogin');
$cookie2 = $_COOKIE['actionAfterLogin']);
var_dump($cookie1);
var_dump($cookie2);
```

outputs:
```
null
string '{"test":1}' (length=10)
```

### PR DESCRIPTION
**Pull request by kidol**
Fixes #9158

I don't understand the reasoning for doing the `cookieValidationKey`-check in 2 different methods. So I added this check to `Request::init()` instead.

### CODE REVIEW

File: framework/web/Response.php [hunk 1]
```diff
@@ -364,16 +364,10 @@ protected function sendCookies()
...
-            if ($request->cookieValidationKey == '') {
-                throw new InvalidConfigException(get_class($request) . '::cookieValidationKey must be configured with a secret key.');
-            }
-            $validationKey = $request->cookieValidationKey;
-        }
         foreach ($this->getCookies() as $cookie) {
             $value = $cookie->value;
-            if ($cookie->expire != 1  && isset($validationKey)) {
-                $value = Yii::$app->getSecurity()->hashData(serialize([$cookie->name, $value]), $validationKey);
+            if ($cookie->expire != 1 && $request->enableCookieValidation && !in_array((string)$cookie->name, $request->cookieValidationWhitelist, true)) {
```
Comment (SamMousa) [inline framework/web/Response.php]: What is the use of strict check combined with a manual type cast?
This is effecitvely the same as using a non-strict check.

Comment (samdark) [inline framework/web/Response.php]: It is.

**Review by cebe — CHANGES_REQUESTED** [review]
> I don't understand the reasoning for doing the cookieValidationKey-check in 2 different methods. So I added this check to Request::init() instead.

If both methods are never called, there is no reason to configure cookie validation key.

**Comment by cebe** [comment]
@kidol are you willing to work on this further?

**Comment by samdark** [comment]
Closing since it seems @kidol isn't going to pick it up again.

**Pull request closed without merging**